enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -2317,23 +2419,19 @@ Function *CreatePrimalAndGradient(
     for (auto &I : *BB)
       unnecessaryInstructionsTmp.insert(&I);
   }
+  CacheAnalysis CA(AA, gutils->oldFunc, gutils->OrigLI, TLI,
+                   unnecessaryInstructionsTmp, _uncacheable_argsPP, topLevel);
   const std::map<CallInst *, const std::map<Argument *, bool>>
       uncacheable_args_map =
           (augmenteddata) ? augmenteddata->uncacheable_args_map
-                          : compute_uncacheable_args_for_callsites(
-                                gutils->oldFunc, gutils->DT, TLI,
-                                unnecessaryInstructionsTmp, AA,
-                                _uncacheable_argsPP, topLevel);
+                          : CA.compute_uncacheable_args_for_callsites();
 
   const std::map<Instruction *, bool> can_modref_map =
-      augmenteddata
-          ? augmenteddata->can_modref_map
-          : compute_uncacheable_load_map(gutils->oldFunc, AA, TLI,
-                                         unnecessaryInstructionsTmp,
-                                         _uncacheable_argsPP, topLevel);
+      augmenteddata ? augmenteddata->can_modref_map
+                    : CA.compute_uncacheable_load_map();
   gutils->can_modref_map = &can_modref_map;
 
-  gutils->forceContexts();
+  // gutils->forceContexts();
```

Comment:
rm

Reply:
I'd like to preserve this for the time being. It does seem reasonable that we can eliminate it and rely on the loop bounds to be generated laziliy, but if later we find this not to be the case it would be nice to see this as the place to return to.